addons/website/static/src/components/dialog/seo.js

```diff
@@ -268,6 +268,17 @@ class TitleDescription extends Component {
         this.maxRecommendedDescriptionSize = 300;
         this.minRecommendedDescriptionSize = 50;
 
+        // Slugify given URL
+        useEffect(() => {
+            if (this.seoContext.seoName) {
+                // `NFKD` as in `http_routing` python `slugify()`
+                this.seoContext.seoName = this.seoContext.seoName.trim().normalize('NFKD').toLowerCase()
+                    .replace(/\s+/g, '-') // Replace spaces with -
+                    .replace(/[^\w-]+/g, '-') // Remove all non-word chars
+                    .replace(/-\+/g, '-'); // Replace multiple - with single -
+            }
+        }, () => [this.seoContext.seoName]);
```

Comment:
I'm not 100% sure but doesn't this mean you'll update the name twice?
- User inputs value
- State update triggers the useEffect
- useEffect updates that state
- State update triggers the useEffect
- It stops here since the value will be identical

In this case it might be worth using the onInput event instead of the [OWL-QWeb shortcut t-model](https://github.com/odoo/owl/blob/master/doc/reference/input_bindings.md)

Note that they already offer are .trim. But in this case since you need to do more than just update the state and instead compute the value with regexps it might be worth using an on-input house made, so that you can treat the value from the event right away instead of relaying on the state update.

I'm not sure if that's a best practice though, what do you think?

Reply:
Just checked, and if you enter a special character that's escaped, it will indeed update the state twice.
Another thing, though it might just be an IMP instead of a fix since it's already like this in 15.0 (didn't check earlier), is that we cannot enter `-` if the character is alone, but if we enter -o by pressing them together, it will work. Maybe we should make the input lazy or something similar?

Reply:
Thanks for the guidance.
Change [applied](https://github.com/odoo/odoo/compare/4228463876d78951605c857e01f19dda3584c959..190b6d05b0a9f6ac7a898d6860d66b237f5fc584).

> Another thing, though it might just be an IMP instead of a fix since it's already like this in 15.0 (didn't check earlier), is that we cannot enter - if the character is alone, but if we enter -o by pressing them together, it will work

It's designed that way, the last char can't be a `-`, so if you type only `-`, it is considered as the last char and can't be typed. If that's what you were referring to :)